src/CodeGen_Hexagon.cpp

```diff
@@ -344,6 +480,16 @@ void CodeGen_Hexagon::compile_func(const LoweredFunc &f,
     debug(1) << "Optimizing Hexagon instructions...\n";
     body = optimize_hexagon_instructions(body, target, alignment_info);
 
+    // Add Hexagon sysmon markers.
+    if (target.has_feature(Target::HVX_sysmon)) {
```

Comment:
Why both the target feature and the environment variable? Seems like we could do this with just the environment variable, which defaults to 0, which means do nothing.

I think the bar for target features should be pretty high because they have long term support implications. They can't be changed/removed easily without breaking code. I'm not sure this meets that bar especially given there's this immediately nested additional condition (on the environment variable).